tests/parametric/test_span_links.py

```diff
@@ -15,56 +15,68 @@
 class Test_Span_Links:
     @staticmethod
     def _get_span_links(span):
-        if context.library >= "python@2.4.0":
-            # trace API v0.5
-            encoded_span_links = span.get("meta", {}).get("_dd.span_links")
-            if not encoded_span_links:
-                return None
-
-            return json.loads(encoded_span_links)
-
-        # trace API v0.4
-        return span.get("span_links")
-
-    @bug(library="python", reason="Since using v0.5 API, trace_id are differents")
+        if span.get("span_links"):
+            # trace API v0.4
+            return span["span_links"]
+        # trace API v0.5
+        encoded_span_links = span.get("meta", {}).get("_dd.span_links")
+        if not encoded_span_links:
+            return None
+
+        span_links = json.loads(encoded_span_links)
+        # convert v0.5 SpanLinks trace IDs to the v0.4 format. This will simplify tests
+        for link in span_links:
+            tid_high = int(link["trace_id"][:16], 16)
+            if tid_high:
+                link["trace_id_high"] = tid_high
+            link["trace_id"] = int(link["trace_id"][16:], 16)
+            link["span_id"] = int(link["span_id"], 16)
+        return span_links
+
+    @pytest.mark.parametrize("library_env", [{"DD_TRACE_API_VERSION": "v0.5"}, {"DD_TRACE_API_VERSION": "v0.4"}])
     def test_span_started_with_link(self, test_agent, test_library):
         """Test adding a span link created from another span.
         This tests the functionality of "create a direct link between two spans
         given two valid span (or SpanContext) objects" as specified in the RFC.
         """
         with test_library:
-            with test_library.start_span("root") as parent:
-                with test_library.start_span(
-                    "child",
-                    parent_id=parent.span_id,
-                    links=[Link(parent_id=parent.span_id, attributes={"foo": "bar", "array": ["a", "b", "c"]})],
-                ):
-                    pass
-
-        traces = test_agent.wait_for_num_traces(1)
-        trace = find_trace_by_root(traces, Span(name="root"))
-        assert len(trace) == 2
+            with test_library.start_span("first") as first:
+                pass
 
-        root = root_span(trace)
-        span = find_span(trace, Span(name="child"))
+            first_span_link = Link(
```

Comment:
This change break the test.
This test was supposed to create a span links between to span instances.
Now, you only test the span link addition. The span link is created within the test, not the client app.

Reply:
Actually this test still creates a link between two spans in the client app. The main change here is how the link is sent to the client app. Instead of just sending a parent_id and attributes to the client app this test sends all the fields that can be used to create a span link. This makes the test easier to read/maintain.

This test aims to verify that span links are serialized in the expected format. I think that's the key element we need to test.